### PR TITLE
[Enhancement] Result view

### PR DESCRIPTION
**Pull request by schultek**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

Improve the visual presentation of feedbacks.

### Description

- Feedback items are now displayed based on their type (Test Case / SCA Issue / Manual Feedback) and their credits.
- Feedback items show their credits, if they have some.
- Students do not see the test case names and have no distinction between automatic and manual Feedback. (All is labeled 'Feedback')
- Instructors see test case names, positive test cases and have a distinction between test case and manual feedback. (Labeled either Test Case or Tutor)
- A score bar is added on top of the feedback, showing achieved points and deductions
- A 'Preliminary' badge is shown if the result is of type Automatic, because of potential hidden test cases.

### Steps for Testing

Check the feedback view for different exercises and both as student and instructor. The feedback items should be shown as described above.

1. Test for a Programming Exercise without manual feedback.
2. Test for a Programming Exercise with manual feedback.
3. Test for the other exercise types that the results and feedbacks still display correctly.

### Test Coverage

**result-detail.component.ts**: 94.44% stmts

### Screenshots

**Student View**
Bildschirmfoto 2020-10-26 um 02 23 14
**Instructor View**
Bildschirmfoto 2020-10-26 um 01 55 33

### CODE REVIEW

**Comment by krusche** [comment]
The screenshots look quite nice already! Do you also support exercises with bonus? (i.e. more than 100%)

**Comment by krusche** [comment]
make sure to add some tests before setting the PR to ready for review and to include the test coverage statistics

**Review by tiffca — DISMISSED** [review]
The result bar for the programming exercises displayed as described. 

Tested on test server 1 programming with/without SCA, auto, and manual, and all scenarios functioned as described. 

The points seem to add up/display as expected when SCA/bonus grading is enabled, manual feedback overrides as expected. 

I was able to view the preliminary indicator if build plan date is set. 
image

tutor feedback is displayed to instructors only. scenarios make sense, for example in a scenario with auto, SCA, and manual grading. 
image

the deductions display and make sense
image

with bonus
image


everything worked as described. 

The feedback for text, model, and file upload seem unchanged (no result bar is shown), I think this was the reason for testing these? for example, no result bar shown to instructor or students. 
image

**Review by fde312 — CHANGES_REQUESTED** [review]
I really like how you have implemented it 👍 
Just a small remark, could you change the color of the manual result with score of 0 to yellow, to keep it consistent? You can then use the grey for the SCA :)

**Review by kloessst — APPROVED** [review]
Looks really nice. Suggestions by @fde312 have been integrated. Code looks good too. Well done 💯

**Review by MaisiKoleni — APPROVED** [review]
Looks excellent and works really well. Displaying the positive test cases is a great addition.

We should have a look how more monospaced font dependent error messages look thought. But I don't have a solution for that right now without making all monospaced.

File: src/main/webapp/app/exercises/shared/result/result-detail.component.ts [hunk 1]
```diff
@@ -129,4 +217,104 @@ export class ResultDetailComponent implements OnInit {
...
+        }
+    }
+
+    /**
+     * Handles the coloring of each feedback items based on its type and credits.
+     * @param feedback The feedback item
+     */
+    getClassNameForFeedbackItem(feedback: FeedbackItem): string {
+        if (feedback.type === FeedbackItemType.Issue) {
+            return 'alert-warning';
```
Comment (fde312) [inline src/main/webapp/app/exercises/shared/result/result-detail.component.ts:256]: Now we have both the SCA and the manual feedback with score 0 in yellow. I think this is okay for now. If we encounter some misunderstandings from the students side, we can adjust it in a follow up  :)

**Review by fde312 — APPROVED** [review]
Code looks good as well. A really cool enhancement! 👍 🥇

**Pull request merged**